### PR TITLE
Reduce complexity of `ProductSection`

### PR DESCRIPTION
**Pull request by ardelato**
### Description

This came about while trying to create tests for #452. At first glance, it is very hard to associate the rendered components on a product page with the components defined in [`ProductSection`](https://github.com/iFixit/react-commerce/blob/main/frontend/templates/product/sections/ProductSection/index.tsx)

This pull aims at trying to reduce the complexity of `ProductSection` and make it easier to read.

## Changes:
- Extracted the Product description section as its own component
- Extracted the compatible devices section as its own component
- Reused the `WikiHtmlAccordianItem` component for the `Specifications` section 
- Added whitespace between sibling components to differentiate where one component ends and another begins
- Moved the zoomed image component to the top of the component tree

## QA Notes:
- [ ] Confirm the Product page and its sections still look the same as before
	- [ ] Product description section is rendered
		- [ ] The `note`, `disclaimer`, and `warning` sections are still rendered if the product has them
	- [ ] Product compatible devices section is rendered if available
		- [ ] The `See all compatible devices` section still renders when expected
	- [ ] Product specifications section is rendered if available
	- [ ] The zoomed-in image on hover still renders

**cc:** @dhmacs 
cr_req 2

### CODE REVIEW

**Comment by k0rvusk0r4x** [comment]
- The zoom in continues to work.
- Description, Kit contents, Compatibility, and Specifications collapsible sections are default expanded or not the same as main and continue to work. 
- Replacement guide section looks and works the same as main.
- Frequently bought together looks and works the same as main.
- Reviews works and looks the same as main.
- Compatibility section renders properly with the correct information.
- Featured Products section works and looks the same as main.
- The colorful boxes (`note`, `disclaimer`, and `warning`) look as on main and collapse with the rest of the description as expected.

Products used:

- Compatible with several devices; `note`, `disclaimer`, and `warning`: https://react-commerce-prod-pebb3grbj-ifixit.vercel.app/products/macbook-air-13-late-2010-2017-replacement-battery
- Compatible with one device, purchase limit warning: https://react-commerce-prod-pebb3grbj-ifixit.vercel.app/products/google-pixel-6-pro-screen-genuine
- Compatible with one device, no warnings: https://react-commerce-prod-pebb3grbj-ifixit.vercel.app/products/steam-deck-fan
- Tool with warning: https://react-commerce-prod-pebb3grbj-ifixit.vercel.app/products/iopener
- Tool with no warnings: https://react-commerce-prod-pebb3grbj-ifixit.vercel.app/products/essential-electronics-toolkit

Tested on:

- Chrome
- Firefox
- Safari
- iPhone
- Android

QA 😺

File: frontend/templates/product/sections/ProductSection/CompatibleDevices.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+   const compatibilityDrawerModelsTruncate = 4;
+   const compatibilityDrawerDeviceTruncate = 3;
+   const compatibilityDrawerIncomplete =
+      product.compatibility &&
+      (product.compatibility.devices.length >
+         compatibilityDrawerDeviceTruncate ||
+         product.compatibility.devices.some(
+            (currentValue) =>
+               currentValue.variants.length > compatibilityDrawerModelsTruncate
+         ));
```
Comment (dhmacs) [inline frontend/templates/product/sections/ProductSection/CompatibleDevices.tsx:20]: This code might be improved a little bit

```suggestion
const MAX_VISIBLE_MODELS_PER_DEVICE = 4;

const MAX_VISIBLE_DEVICES = 3;

export function CompatibleDevices({ product }: CompatibleDevicesProps) {
  const deviceCount = product.compatibility?.devices.length ?? 0;
  
  const hasMoreDevicesToShow = 
    deviceCount > MAX_VISIBLE_DEVICES ||
    product.compatibility?.devices.some(
       (device) => device.variants.length > MAX_VISIBLE_MODELS_PER_DEVICE
    );
```

**Review by dhmacs — APPROVED** [review]
Good stuff @ardelato 👏 

I've left an optional suggestion, but no need to do it now

CR 🏎️

**Comment by ardelato** [comment]
deploy_block ⚡ I'm going to take a lot at applying some changes from the suggestions

**Comment by ardelato** [comment]
un_deploy_block 🎲  I applied the suggestions to `CompatibleDevices`

I also updated the **QA Notes** to test that the `See all compatible devices` continues to work.

**Review by danielbeardsley — APPROVED** [review]
CR :+1: 


Thanks!

**Comment by erinemay** [comment]
dev_block 🦺 
The image is showing up before product details on mobile.
Confirmed on Android + Chrome and iOS + Safari

<details><summary>Screenshots</summary>

| Main | Branch |
| - | - |
| screenshot_20221209-094431 | screenshot_20221209-094417 |

</details>

Otherwise, works same as https://github.com/iFixit/react-commerce/pull/1108#issuecomment-1341380434

**Comment by ardelato** [comment]
Nice find @erinemay! I reverted the commit where we move the zoomed imaged component to the top of the component tree, which looks like it fixes the issue.

un_dev_block ♻️

**Comment by deltuh-vee** [comment]
dev_block ✊ 
Clicking to show zoom is broken on mid sized mobile devices(Tested on iPad mini 5th gen). The same zoom image is shown regardless of where I click. On master, zoom is still working and the zoomed image changes depending on where I tap.
IMG_5D314B91F63D-1

**Comment by ardelato** [comment]
un_dev_block ⚡ spoke with @deltuh-vee and the webpage was in a weird state. Refreshing the page fixed the issue.

**Comment by deltuh-vee** [comment]
QA 🎬 
Product details now show up before the product image and the product page still works as in https://github.com/iFixit/react-commerce/pull/1108#issuecomment-1341380434.

**Review by mlahargou — APPROVED** [review]
CR 📱 

Nice! I think testing will be easier!

**Comment by jordycosta** [comment]
### Testing Pulldasher-Dev

**Pull request merged**